packages/compiler/lib/decorators.ts

```diff
@@ -454,11 +454,11 @@ const tagPropertiesKey = Symbol();
 // Set a tag on an operation or namespace.  There can be multiple tags on either an
 // operation or namespace.
 export function $tag(program: Program, target: Type, tag: string) {
-  if (target.kind !== "Operation" && target.kind !== "Namespace") {
+  if (target.kind !== "Operation" && target.kind !== "Namespace" && target.kind !== "Interface") {
```

Comment:
The only issue was with validating the type? It already worked to crawl parent? guessing it is from the test

Reply:
Yep, it only grabs tags from the closest parent, doesn't walk the tree.  That functionality can also be added but James didn't do it at the time.

Reply:
so if you have 

```
@tag("xzy")
namespace Foo {
   interface Bar {
     op f();
   }
}
```

xyz doesn't get added to `f`?

Reply:
As of right now it doesn't, but I can add it really quickly if you like

Reply:
yeah, I feel like it is an weird limitation to only go up to the parent.